microcosm_flask/errors.py

```diff
@@ -0,0 +1,111 @@
+"""
+Generalized error handling.
+
+"""
+from logging import getLogger
+
+from flask import jsonify
+from werkzeug.exceptions import default_exceptions
+
+
+error_logger = getLogger("errors")
+
+
+def extract_status_code(error):
+    """
+    Extract an error code from a message.
+
+    """
+    try:
+        return error.code
+    except AttributeError:
+        try:
+            return error.status_code
+        except AttributeError:
+            try:
+                return error.errno
```

Comment:
yep. one thought - might be OK to propagate by default if debug/testing is True (either as the returned status code or as part of some error context field in the error JSON response)